lib/runtime/wasmer/imports_test.go

```diff
@@ -249,6 +256,174 @@ func Test_ext_storage_set_version_1(t *testing.T) {
+	numKps := 5
+
+	for i := 0; i < numKps; i++ {
+		kp, err := ed25519.GenerateKeypair()
+		if err != nil {
+			t.Fatal(err)
+		}
+		inst.inst.ctx.Keystore.Insert(kp)
+		expectedKey := kp.Public().Encode()
+		expectedPubKeys = append(expectedPubKeys, expectedKey)
+	}
+
+	idData := []byte{2, 2, 2, 2}
+
+	ptr, err := inst.inst.malloc(uint32(len(idData)))
+	require.NoError(t, err)
+
+	inst.inst.store(idData, int32(ptr))
+	dataLen := int32(len(idData))
+
+	require.NoError(t, err)
+
+	runtimeFunc, ok := inst.inst.vm.Exports["rtm_ext_crypto_ed25519_public_keys_version_1"]
+	require.True(t, ok)
+
+	out, err := runtimeFunc(int32(ptr), dataLen)
+	require.NoError(t, err)
+
+	mem := inst.inst.vm.Memory.Data()
+	//resultLenBytes := mem[0:32]
+	//resultLen := binary.LittleEndian.Uint32(resultLenBytes)
+	pubKeyData := mem[out.ToI32() : out.ToI32()+32]
+	fmt.Println("pubKeyData: ", pubKeyData)
+	pubKeys := [][]byte{}
+	for i := 0; i < numKps; i++ {
+		kpData := mem[out.ToI32()+int32(i*32) : out.ToI32()+int32((i+1)*32)]
+		pubKeys = append(pubKeys, kpData)
+	}
+
+	sort.Slice(expectedPubKeys, func(i, j int) bool { return bytes.Compare(expectedPubKeys[i], expectedPubKeys[j]) < 0 })
+	sort.Slice(pubKeys, func(i, j int) bool { return bytes.Compare(pubKeys[i], pubKeys[j]) < 0 })
+
+	require.Equal(t, expectedPubKeys, pubKeys)
+}
+
+func Test_ext_crypto_ed25519_sign_version_1(t *testing.T) {
+	inst := NewTestInstance(t, runtime.HOST_API_TEST_RUNTIME)
+	//mem := inst.inst.vm.Memory.Data()
+
+	kp, err := ed25519.GenerateKeypair()
+	require.Nil(t, err)
+
+	inst.inst.ctx.Keystore.Insert(kp)
+
+	idData := []byte{2, 2, 2, 2}
+
+	idPtr, err := inst.inst.malloc(uint32(len(idData)))
+	require.NoError(t, err)
+
+	inst.inst.store(idData, int32(idPtr))
+
+	pubKeyData := kp.Public().Encode()
+	msgData, err := common.HexToBytes("0xce0677bb30baa8cf067c88db9811f4333d131bf8bcf12fe7065d211dce971008")
+	require.NoError(t, err)
+
+	params := append(pubKeyData, msgData...)
+
+	paramPtr, err := inst.inst.malloc(uint32(len(params)))
+	require.NoError(t, err)
+
+	inst.inst.store(params, int32(paramPtr))
+
+	runtimeFunc, ok := inst.inst.vm.Exports["rtm_ext_crypto_ed25519_sign_version_1"]
+	require.True(t, ok)
+
+	ret, err := runtimeFunc(int32(idPtr), int32(paramPtr))
+	require.NoError(t, err)
+
+	fmt.Println(ret.ToI32())
+}
+
+func Test_ext_crypto__ecdsa_recover_version_1(t *testing.T) {
+	inst := NewTestInstance(t, runtime.HOST_API_TEST_RUNTIME)
+
+	msgData := []byte("Hello world!")
+	blakeHash, err := common.Blake2bHash(msgData)
+	require.Nil(t, err)
+
+	pubKey, secKey := generateKeyPairs()
+	sigData, err := secp256k1.Sign(blakeHash.ToBytes(), secKey)
+	require.NoError(t, err)
+
+	fmt.Println(pubKey)
+
+	encSign, err := scale.Encode(sigData)
+	require.NoError(t, err)
+	encMsg, err := scale.Encode(blakeHash.ToBytes())
+	require.NoError(t, err)
+
+	ret, err := inst.Exec("rtm_ext_crypto_secp256k1_ecdsa_recover_version_1", append(encSign, encMsg...))
+
+	buf := &bytes.Buffer{}
```

Comment:
@noot Can you check this test? It's returning two extra bytes. In Wasm the length is set to 68. But returning 70 here. `buf.Write(ret[2:])` This fixes the test. But any idea about this strange behaviour?